templates/telemetry_derived/bq_events/query.sql

```diff
@@ -0,0 +1,139 @@
+-- Creates a backwards-compatible flattened events dataset created from gcp-ingested pings
+-- Designed to provide continuity for the `events` dataset via a user-facing union view
+WITH event_events AS (
+  SELECT
+    *,
+    e,
+    "content" AS event_process
+  FROM
+    telemetry.event
+  CROSS JOIN
+    UNNEST(payload.events.content) AS e
+  UNION ALL
+  SELECT
+    *,
+    e,
+    "dynamic" AS event_process
+  FROM
+    telemetry.event
+  CROSS JOIN
+    UNNEST(payload.events.dynamic) AS e
+  UNION ALL
+  SELECT
+    *,
+    e,
+    "extension" AS event_process
+  FROM
+    telemetry.event
+  CROSS JOIN
+    UNNEST(payload.events.extension) AS e
+  UNION ALL
+  SELECT
+    *,
+    e,
+    "gpu" AS event_process
+  FROM
+    telemetry.event
+  CROSS JOIN
+    UNNEST(payload.events.gpu) AS e
+  UNION ALL
+  SELECT
+    *,
+    e,
+    "parent" AS event_process
+  FROM
+    telemetry.event
+  CROSS JOIN
+    UNNEST(payload.events.parent) AS e
```

Comment:
I think it would be better to `UNNEST` like this:
```sql
SELECT
  event.*,
  e,
  event_process
FROM
  telemetry.event AS event
CROSS JOIN
  UNNEST([
    STRUCT("content" AS event_process, payload.events.content as events),
    ("dynamic", payload.events.dynamic),
    ("extension", payload.events.extension),
    ("gpu", payload.events.gpu),
    ("parent", payload.events.parent)
  ])
CROSS JOIN
  UNNEST(events) AS e
```

Reply:
Oh, lovely – didn't even realize this was possible :D